proxy/src/main/java/com/velocitypowered/proxy/protocol/ProtocolUtils.java

```diff
@@ -194,12 +203,93 @@ public static void writeProperties(ByteBuf buf, List<GameProfile.Property> prope
     return properties;
   }
 
+  private static final int FORGE_MAX_ARRAY_LENGTH = Integer.MAX_VALUE & 0x1FFF9A;
+
+  /**
+   * Reads an byte array for legacy version 1.7 from the specified {@code buf}
+   *
+   * @param buf the buffer to read from
+   * @return the read byte array
+   */
+  public static byte[] readArray17(ByteBuf buf) {
+    // Read in a 2 or 3 byte number that represents the length of the packet. (3 byte "shorts" for
+    // Forge only)
+    // No vanilla packet should give a 3 byte packet
+    int len = readVarShort(buf);
+
+    Preconditions.checkArgument(len <= (FORGE_MAX_ARRAY_LENGTH),
+        "Cannot receive array longer than %s (got %s bytes)", FORGE_MAX_ARRAY_LENGTH, len);
+
+    byte[] ret = new byte[len];
+    buf.readBytes(ret);
+    return ret;
+  }
+
+  /**
+   * Writes an byte array for legacy version 1.7 to the specified {@code buf}
+   *
+   * @param b array
+   * @param buf buf
+   * @param allowExtended forge
+   */
+  public static void writeArray17(byte[] b, ByteBuf buf, boolean allowExtended) {
+    if (allowExtended) {
+      Preconditions
+          .checkArgument(b.length <= (FORGE_MAX_ARRAY_LENGTH),
+              "Cannot send array longer than %s (got %s bytes)", FORGE_MAX_ARRAY_LENGTH,
+              b.length);
+    } else {
+      Preconditions.checkArgument(b.length <= Short.MAX_VALUE,
+          "Cannot send array longer than Short.MAX_VALUE (got %s bytes)", b.length);
+    }
+    // Write a 2 or 3 byte number that represents the length of the packet. (3 byte "shorts" for
+    // Forge only)
+    // No vanilla packet should give a 3 byte packet, this method will still retain vanilla
+    // behaviour.
+    writeVarShort(buf, b.length);
+    buf.writeBytes(b);
+  }
+
+  /**
+   * Reads a Minecraft-style VarInt from the specified {@code buf}.
+   *
+   * @param buf buf to write
+   * @return read int
+   */
+  public static int readVarShort(ByteBuf buf) {
```

Comment:
This isn't really a `VarShort`, since it can encode higher numbers than a short. This should get a different name, maybe `readExtendedShort` (reflects the javadoc of forge 1.7.10), or `readExtendedForgeShort` (because it's only used by forge anyway).
The same applies to `writeVarShort`.